app/controllers/ncr/work_orders_controller.rb

```diff
@@ -74,5 +71,50 @@ def add_approvals
         @model_instance.setup_approvals_and_observers(@approver_email)
       end
     end
+
+    ## update helpers ##
+    # TODO move into a standalone class
+
+    def amount_increased?
+      changes = @model_instance.previous_changes
+      changes.key?('amount') && (@model_instance.amount > changes['amount'].first)
+    end
+
+    def budget_codes_changed?
+      changes = @model_instance.previous_changes
+      Ncr::WorkOrder.budget_code_fields.any? do |field|
+        changes.key?(field.to_s)
+      end
+    end
+
+    def budget_approver?
+      @model_instance.budget_approvers.include?(current_user)
+    end
+
+    def requires_budget_reapproval?
+      @model_instance.approved? && (
+        self.amount_increased? || (
+          self.budget_codes_changed? &&
+          !self.budget_approver?
+        )
+      )
+    end
+
+    def reapprove_if_necessary
+      if requires_budget_reapproval?
+        @model_instance.restart_budget_approvals
+        flash[:success] = "Successfully modified! This request now needs to be re-approved by budget."
+      end
+    end
+
+    def after_update
+      if @model_changing && !@model_instance.emergency # skip approvals if emergency
```

Comment:
I thought `emergency` didnt matter? or was that a different type of model instance?